### ISSUE TITLE
Product Collection 'Preview' label should not be a button

### ISSUE THREAD

**Issue by Aljullu** [woothemes/woocommerce#49291]
### Describe the bug

I think the `Preview` label shown on the Product Collection block when it's on preview mode has the wrong HTML label. Currently, it's a `<button>` but I think it would be better a `<div>`, `<span>` or something similar. I noticed because it was showing a pointer when hovering it, but clicking on it did nothing. After [inspecting the code](https://github.com/woocommerce/woocommerce/blob/dacee984fadc1e651b5ae85de347849c9a8f8e0d/plugins/woocommerce-blocks/assets/js/blocks/product-collection/edit/product-collection-content.tsx#L138-L149), it looks like there is no `onClick` associated with it, so if it's not clickable, I think it would be better if it wasn't a button. @imanish003 what do you think?

### Steps to reproduce

1. Go to Appearance > Editor > Templates > Product by Category.
2. Select the Product Collection block.
3. Notice that a _Preview_ label appears in the top right.

Actual behavior: hovering the _Preview_ label shows a `pointer` cursor, and the element is clickable (it's a button).

![Captura de pantalla de 2024-07-09 13-22-46](https://github.com/woocommerce/woocommerce/assets/3616980/d6afa9ba-30e9-45bd-b4f1-726f80e0bc77)

Expected behavior: hovering the _Preview_ label should display the `default` cursor (or `text`), and the element should not be clickable (it shouldn't be a button and styles shouldn't change when clicked).

### WordPress Environment

Tested on WP 6.6 and WC `trunk`.

**Comment by imanish003** [comment]
Hey @Aljullu, Thanks for raising this issue :pray:

My reasoning behind using a button was that I could make it work out of the box with the Gutenberg button. The Gutenberg button has a `showTooltip` prop that shows a tooltip when hovered. This was also the same in style as our design. 

But I agree with you—it's not a button, so it shouldn't be a button. We can implement something custom for this, but AFAIK, this can't be achieved using Gutenberg components.

FYI: @kmanijak

**Comment by amesplant** [comment]
I am investigating this one.

**Comment by kmanijak** [comment]
@amesplant , nice, thanks! Let us know how that goes :)

**Comment by amesplant** [comment]
@imanish003 cc @Aljullu @kmanijak 
What is the goal of the 'preview' item? Is there a way to display that information differently?

The editor itself has a number of accessibility issues, such as the inability to use just a keyboard (note that I am unable to tab through the editor window). As an Accessibility Expert, I am not all that concerned that it is a `button` vs `span` and more concerned that using a tooltip isn't the best direction.

**Comment by WunderBart** [comment]
> But I agree with you—it's not a button, so it shouldn't be a button. We can implement something custom for this, but AFAIK, this can't be achieved using Gutenberg components.

> I am not all that concerned that it is a button vs span and more concerned that using a tooltip isn't the best direction.

👋 @ciampo, is there an accessible Components alternative we could use here?

**Comment by ciampo** [comment]
I have little context on this, but here are a few observations:

- the "preview" label definitely shouldn't be a button if it's not interactive; it seems to be more of a description of whatever is being highlighted
- is this pattern of labelling a highlighted block something unique to the product collection?
- using a tooltip there feels hacky, a way to squeeze in more text in a place where there isn't room for it
- in case this should be an interactive piece of UI, the block toolbar or block inspector sidebar feel like a more natural place to me for such UI

Specific to `@wordpress/components`:
- [`Button`](https://wordpress.github.io/gutenberg/?path=/docs/components-button--docs) renders a `button` or an `a` tag, so it's not suitable if the UI being displayed here is not interactive;
- If you need tooltip functionality, you can use the [`Tooltip` component](https://wordpress.github.io/gutenberg/?path=/docs/components-tooltip--docs) even without a button

**Comment by amesplant** [comment]
@imanish003 cc @Aljullu @kmanijak
Let me know what direction you would like me to take. I mostly need to understand the purpose of the `preview` button before I can implement an appropriate solution.

**Comment by kmanijak** [comment]
@amesplant , the "Preview" is purely informational. It's not meant to be interactive (except for showing tooltip with more explanation) but overall it's just information to the merchant that products being displayed in Product Collection block are not necessarily the ones that will be displayed in the frontend.

**Comment by imanish003** [comment]
@amesplant As Karol mentioned, the "Preview" label is purely informational. 

The problem we are trying to solve using the "Preview" label is to inform the merchant that the products displayed in the Product Collection block on the Editor side will not be the same as those displayed on the front end. 

The reason being, sometimes we don't have all the information available on the editor side to display the exact products that will be displayed on the frontend. You may find more info in [this PR](https://github.com/woocommerce/woocommerce/pull/46369) in which this was implemented.

For example, for the "Product by Category" template, we don't know the exact category to show products from, therefore we can't show the exact products on the editor side. On frontend, we will know the category and can show the exact products. Same is the case for "Products by attribute" and "Products by tag" templates.

I hope this helps you understand the purpose of the "Preview" label. Please feel free to ask if you have any more questions 🙂

**Comment by amesplant** [comment]
@kmanijak @imanish003 thanks you for the bigger picture for me, I feel like I have a much better grasp of the "ask", much appreciated. I will investigate a bit more this week to see if we can come up with a good suitable solution for this on the Admin side of the editor. Stay tuned 🙂

**Comment by amesplant** [comment]
@imanish003 cc @Aljullu @kmanijak
As per the PR above, I have updated the 'tooltip button' to be a Notice instead. Let me know how you feel about this approach.

![image](https://github.com/user-attachments/assets/029d7534-c2f4-49df-8ca8-cbb67885783c)

**Comment by imanish003** [comment]
@amesplant I believe @jarekmorawski explored this while designing preview label. I think he can provide more insights on this 🙂 CC: @jarekmorawski

As far as I remember, the preview label was designed to be a button with a tooltip because we want the Editor to be as close as possible to the frontend. We don't want to show anything on the Editor side that won't be shown on the frontend. As you will notice, the preview label is shown only when the Product Collection block is selected on the Editor side. It was done on purpose, i.e., the Editor should be as close as possible to the frontend.

But I can understand your accessibility concerns. Aren't there other ways to fix the accessibility issues without changing the design? Can we figure out a way how to make preview label + tooltip fully accessible with keyboard navigation. Or using aria-label or something similar. ( AFAIK we already have aria-label equal to the tooltip text for the preview label button) 🤔

**Comment by amesplant** [comment]
@imanish003 cc @jarekmorawski 
This should work similarly in the respect that the Notice will **only** appear when the Product Collection block is selected. 

The editor experience is inherently troublesome for users using a keyboard, and so if we can find a solution that does not require the use of a button or a tooltip in the editor it will make it a better experience for all users.

**Comment by imanish003** [comment]
@amesplant One problem I can see with Notice is that it will shift the content below it when it appears and disappears. Isn't this the case? 🤔

As @jarekmorawski designed the preview label, I will wait for his input on this. I think he can provide more insights on this 🙂

> The editor experience is inherently troublesome for users using a keyboard, and so if we can find a solution that does not require the use of a button or a tooltip in the editor it will make it a better experience for all users.

💯

**Comment by jarekmorawski** [comment]
Accessibility is crucial, thanks for raising this issue @amesplant. We went with a badge because it goes against Gutenberg's guidelines to change the layout when the user selects a block (users will see the screen shift if we go with the notice).

Is there anything else we could try? Would it help if we moved the badge inside or closer to the block toolbar?

From the top of my head, we'd move the notice to the bottom of the block so it doesn't cause a visible layout shift. It'd no longer be a button, just a notice that users can dismiss by clicking the 'X' button. It's not the perfect UX because some users may never see the notice on catalog pages with 100s of products... I couldn't figure out a better way that wouldn't force us to reposition the block toolbar or display the notice inside the block's editable area.

![Bottom notice](https://github.com/user-attachments/assets/dc9c448c-5f53-4c31-a6f4-4ee796f92910)

We'd also display the notice in the Inspector instead of the canvas. That'd probably be the simplest and most elegant solution that doesn't break any guidelines. However, it'll make an already complex screen even more overwhelming, so we'd try that as an experiment and see how users feel about it.

![Inspector](https://github.com/user-attachments/assets/053da87a-75ca-49b4-a391-d7d15fd50ec7)

If we are to display the notice at the top, I'd prefer to make it subtler and dismissible. It'd work well in conjunction with a notice in the Inspector.

![Top notice](https://github.com/user-attachments/assets/036f2844-9168-4fd0-8958-11c5e09178d8)

**Comment by imanish003** [comment]
Hey @amesplant,

> so if we can find a solution that does not require the use of a button or a tooltip in the editor it will make it a better experience for all users.

AFAIU If we just replace the button with `div` or `span`, then it will fix the accessibility issue. Is there any other accessibility issue that we need to address? 🤔 
- The button has an `aria-label` that is good for screen readers. It has the correct text: "Actual products will vary depending on the page being viewed."
- Regarding keyboard navigation, After selecting product collection, if I click "tab" twice, I can see the tooltip. So, I think it's accessible. Please see attached video. 

Honestly, I'm not happy with the alternatives, so I'd rather stick with the current design unless it's absolutely necessary. However, if there's a significant accessibility issue, then I'm open to considering the alternatives proposed by @jarekmorawski 🙂

https://github.com/user-attachments/assets/180ca9a9-978a-449a-9207-8ff4ee861066

**Comment by amesplant** [comment]
@imanish003 @jarekmorawski 
Thanks so much for the additional context and based on the fact that having a shift in the layout is problematic, I would concur that either keeping the tooltip or adding the content to the inspector are better options.

The tooltip **should** semantically be a button since a `<div>` and `<span>` are not inherently focusable elements. The tooltip needs to be a focusable element in order for a keyboard user to activate it, and it is OK for the button to not include a click handler. So .. I would agree to leave it as-is or add the messaging in the inspector.

**Comment by WunderBart** [comment]
@jarekmorawski @amesplant @imanish003 

I'm wondering if it's not the same case as the banner above:

<img width="353" alt="Screenshot 2024-07-26 at 12 45 25" src="https://github.com/user-attachments/assets/414c69c8-778b-4f25-9115-0ea0b61d400f">

What if we make a (closeable?) banner like that just above the products? Wouldn't that be consistent with the current UI? It also would not cause the shift as it would always be there (unless closed manually). For example:

<img width="651" alt="Screenshot 2024-07-26 at 12 52 37" src="https://github.com/user-attachments/assets/8e1e74e3-b554-47bf-a423-60de78cd7da0">

**Comment by imanish003** [comment]
Hey @amesplant 👋🏻

> either keeping the tooltip or adding the content to the inspector are better options.
> The tooltip should semantically be a button since a `<div>` and `<span>` are not inherently focusable elements. The tooltip needs to be a focusable element in order for a keyboard user to activate it, and it is OK for the button to not include a click handler.

Aha! TIL that button is accessible for such cases even if it doesn't have a click handler. Thanks for sharing this valuable information. I appreciate it! 🙏

Based on this, I think we should keep the current tooltip design as it is accessible. I am going to close this issue as we don't need to make any changes.

PS: We can always reopen this issue if anyone thinks otherwise 🙇  CC: @Aljullu @kmanijak

**Comment by kmanijak** [comment]
With all that information, should we maybe use the default cursor instead of the pointer? Or that's actually against accessibility rules?

**Comment by imanish003** [comment]
@kmanijak Good point! I don't think it's against accessibility rules. I think it's a good idea to use the default cursor instead of the pointer, as the button is not clickable. The pointer cursor typically indicates an interactive element that can be clicked. If the button does not have a click handler, it might be misleading to use the pointer cursor. But I will wait for @amesplant to confirm this.

**Comment by Aljullu** [comment]
Heads-up that I'm not an accessibility expert, but making some questions just to make sure we take the right decision here:

> The tooltip **should** semantically be a button since a `<div>` and `<span>` are not inherently focusable elements.

What would you think about using the `tabindex` attribute? Do you think that the element being a `<div>` or `<span>` makes it more difficult for users to know it's a hoverable/focusable element?

> The tooltip needs to be a focusable element in order for a keyboard user to activate it, and it is OK for the button to not include a click handler. So .. I would agree to leave it as-is or add the messaging in the inspector.

That was also a TIL for me. Do you have a link expanding on that, @amesplant? Reading the definitions from the HTML Standard and MDN my understanding is that it's expected that buttons should be "activable".

[HTML Standard](https://html.spec.whatwg.org/multipage/form-elements.html#the-button-element):

> The type attribute controls the behavior of the button when it is activated.

From [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button):

> The button HTML element is an interactive element activated by a user with a mouse, keyboard, finger, voice command, or other assistive technology. Once activated, it then performs an action, such as submitting a form or opening a dialog.

I have to say, though, it's a bit hard for me to fully understand if being "activable" is a requirement or simply the most common usecase. :sweat_smile: 

---

In any case, if we go with a button, then I think we should change the text. Buttons are usually expected to be described by a verb that explains what they do when activated. But this is not the case here. Users using a screen reader will be announced that there is a "Preview button" on the page, but that button can't be clicked, which IMO will be confusing and feel like a bug. That's because here we use "Preview" as a noun, not as a verb. In other words, what we mean is "This is a preview" instead of "Click to view a preview", but the fact that it's used in a button makes it feel that it means the latter.

I might be overthinking this, though, what do you folks think?

**Comment by Aljullu** [comment]
What do you folks think about reopening this issue? Even if we stay with the `<button>` element, I think at least we should proceed with what @kmanijak highlighted above about setting the cursor to `default` and, ideally, removing the hover effect?

**Comment by imanish003** [comment]
@Aljullu I think we can reopen this issue. I think for now, we can go with changing the cursor to default. If needed/required, we can make any other changes in the future. Does this sound good to you? 🙂

**Comment by Aljullu** [comment]
Yup, issue reopened. :slightly_smiling_face:

**Comment by imanish003** [comment]
@Aljullu Thanks. Since it was a small change, I quickly created a PR for it. I will assign you as a reviewer 🤝
